test/extended/etcd/vertical_scaling.go

```diff
@@ -54,6 +55,64 @@ var _ = g.Describe("[sig-etcd][Feature:EtcdVerticalScaling][Suite:openshift/etcd
 		err = errors.Wrap(err, "pre-test: timed out waiting for initial cluster state to have 3 running machines and 3 voting members")
 		o.Expect(err).ToNot(o.HaveOccurred())
 
+		cpmsSupported, err := scalingtestinglibrary.PlatformSupportsCPMS(ctx, g.GinkgoT(), cpmsClient)
+		err = errors.Wrap(err, "pre-test: failed to determine if ControlPlaneMachineSet is present")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		if cpmsSupported {
+
+			// TODO: Add cleanup step to recover back to 3 running machines and members if the test fails
+
+			// step 1: delete a running machine to trigger the CPMSO to create a new one to replace it
+			machineName, err := scalingtestinglibrary.DeleteSingleMachine(ctx, g.GinkgoT(), machineClient)
+			o.Expect(err).ToNot(o.HaveOccurred())
+			framework.Logf("Waiting for machine %q to be replaced", machineName)
+
+			memberName, err := scalingtestinglibrary.MachineNameToEtcdMemberName(ctx, oc.KubeClient(), machineClient, machineName)
+			err = errors.Wrapf(err, "failed to get etcd member name for deleted machine: %v", machineName)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			// step 2: wait until the CPMSO scales-up by creating a new machine
+			// We need to check the cpms' status.readyReplicas because the phase of one machine will always be Deleting
+			// so we can't use EnsureMasterMachinesAndCount() since that looks for non-Deleting machines
+			err = scalingtestinglibrary.EnsureReadyReplicasOnCPMS(ctx, g.GinkgoT(), 4, cpmsClient)
```

Comment:
Seems like we should watch for `replicas` instead of `readyReplicas`. The latter may never surge to 4 in this case as the deleted machine's member can get removed before we scale up.

From https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27497/pull-ci-openshift-origin-master-e2e-aws-ovn-etcd-scaling/1603457840947662848


Member removed at `19:19:21`
```
2022-12-15T19:19:21.221655410Z I1215 19:19:21.221269       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"16340e00-06fa-4526-9dc4-3da772936f35", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'MemberRemove' removed member with ID: 2485787822400865737
2022-12-15T19:19:21.221655410Z I1215 19:19:21.221529       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"16340e00-06fa-4526-9dc4-3da772936f35", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ScaleDown' successfully removed member: [ url: 10.0.182.231, name: ip-10-0-182-231.ec2.internal, id: 2485787822400865737 ] from the cluster
```
Member added as learner at `19:24:50`:
```
2022-12-15T19:24:50.597610435Z I1215 19:24:50.597552       1 event.go:285] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"16340e00-06fa-4526-9dc4-3da772936f35", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'MemberAddAsLearner' successfully added new member https://10.0.152.48:2380
```

Looking back into the clustermemberremoval controller and in particular the PR(https://github.com/openshift/cluster-etcd-operator/pull/947) to relax the scale-down constraint for unhealthy members where I think we may have inadvertently relaxed this for a healthy member too. Need to confirm.

Reply:
Waiting for the fix to not allow scale-down or removal of the machine pending deletion before the new one is added https://github.com/openshift/cluster-etcd-operator/pull/978